### PR TITLE
fix(storage): custom client base64 encoded SSE-C headers

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

* base64 encode `x-amz-server-side-encryption-customer-key` and `x-amz-server-side-encryption-customer-key-md5` headers per: [the requirements](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ServerSideEncryptionCustomerKeys.html#specifying-s3-c-encryption)

#### Description of how you validated changes

* Unit test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/src/AwsClients/S3/utils/base64/index.browser.ts
Comment (AllanZhengYP) [inline packages/storage/src/AwsClients/S3/utils/base64/index.browser.ts:1]: Can you put this into the `packages/storage/src/AwsClients/S3/runtime` folder and reexport from the `/utils`? This is an existing pattern in S3 client

File: packages/storage/src/AwsClients/S3/utils/base64/index.browser.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	const base64Str = Array.from(bytes, x => String.fromCodePoint(x)).join('');
+	return btoa(base64Str);
+}
+
+function base64ToBytes(base64Encoded: string): Uint8Array {
+	const decoded = atob(base64Encoded);
+	return Uint8Array.from(decoded, m => m.codePointAt(0));
+}
+
+export function toBase64(str: string): string {
```
Comment (AllanZhengYP) [inline packages/storage/src/AwsClients/S3/utils/base64/index.browser.ts]: Similar to [AWS SDK](https://github.com/aws/aws-sdk-js-v3/blob/172b4f3475370b386271750436004706b36bc056/packages/middleware-ssec/src/index.ts#L36-L53) we should support the TypedArray/ArrayBuffer view as input.

**Review by AllanZhengYP — APPROVED** [review]
💯 The change looks good to me. Tho I'm not sure what's causing other non-related packages' bundle size test to fail. We need to investigate it.

File: packages/storage/src/AwsClients/S3/utils/serializeHelpers.ts [hunk 2]
```diff
@@ -25,13 +26,28 @@ interface ObjectSsecOptions {
...
 		'x-amz-server-side-encryption-customer-algorithm':
 			input.SSECustomerAlgorithm,
-		'x-amz-server-side-encryption-customer-key': input.SSECustomerKey,
-		'x-amz-server-side-encryption-customer-key-md5': input.SSECustomerKeyMD5,
+		// base64 encoded is need
+		// see: https://docs.aws.amazon.com/AmazonS3/latest/userguide/ServerSideEncryptionCustomerKeys.html#specifying-s3-c-encryption
+		'x-amz-server-side-encryption-customer-key':
+			input.SSECustomerKey && toBase64(input.SSECustomerKey),
+		'x-amz-server-side-encryption-customer-key-md5':
+			input.SSECustomerKeyMD5 && toBase64(sseCustomKeyMD5Digest),
```
Comment (AllanZhengYP) [inline packages/storage/src/AwsClients/S3/utils/serializeHelpers.ts:48]: Can you add a TODO comment here indicating we need to remove the MD5 input parameter in v6?

**Review by AllanZhengYP — APPROVED** [review]
@HuiSF Nice work addressing all the issues so quickly!

**Review by kvramyasri7 — APPROVED** [review]

**Pull request merged**